### PR TITLE
FACES-2999 Develop tests for the Data category of the Alloy Showcase

### PR DESCRIPTION
**Pull request by pcwhite**

### CODE REVIEW

File: jsf-showcase-webapp/pom.xml [hunk 1]
```diff
@@ -238,7 +238,7 @@
 		<dependency>
 			<groupId>com.liferay.faces.test</groupId>
 			<artifactId>com.liferay.faces.test.selenium</artifactId>
-			<version>0.4.0</version>
+			<version>0.4.1-SNAPSHOT</version>
```
Comment (stiemannkj1) [inline jsf-showcase-webapp/pom.xml:241]: Why did you need to depend on a `SNAPSHOT` here? Why is `0.4.0` not adequate?

File: jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java [hunk 2]
```diff
@@ -212,7 +254,8 @@ protected void navigateToUseCase(BrowserDriver browserDriver, String componentPr
 	 * Click the link and assert that it opens a new window/tab with the correct domain name.
 	 */
 	protected void testLink(BrowserDriver browserDriver, WaitingAsserter waitingAsserter, String exampleLinkXpath,
-		String domainNameRegex) {
+
+			String domainNameRegex) {
```
Comment (stiemannkj1) [inline jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java:258]: Seems like a formatting error here.

File: jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java [hunk 3]
```diff
@@ -44,7 +45,7 @@
 public class TesterBase extends BrowserDriverManagingTesterBase {
 
 	// Logger
-	private static final Logger logger = LoggerFactory.getLogger(TesterBase.class);
+	protected static final Logger logger = LoggerFactory.getLogger(TesterBase.class);
```
Comment (stiemannkj1) [inline jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java:48]: Usually we make loggers `private`. If you need to use a logger in a subclass, create a new one. I believe that makes logs easier to read. @ngriffin7a, may correct me though.

File: jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java [hunk 4]
```diff
@@ -132,6 +136,44 @@ protected String capitalize(String string) {
 		return capitalizedString;
 	}
 
+	/**
+	 * Click an option and wait for Ajax to rerender the option. This method exists because {@link
+	 * BrowserDriver#clickElementAndWaitForRerender(java.lang.String)} does not work on selectOneMenu,
+	 * selectManyListbox, and SelectManyMenu. For more information see method comments.
+	 */
+	protected void clickOptionAndWaitForRerender(BrowserDriver browserDriver, String optionXpath) {
```
Comment (stiemannkj1) [inline jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java:144]: This seems like a nice change.

File: jsf-showcase-webapp/pom.xml (hunk 1 above)
Comment (pcwhite) [reply jsf-showcase-webapp/pom.xml:241]: @stiemannkj1 It's 0.4.1-SNAPSHOT in liferay-faces-test-selenium, and I wasn't sure if there would be any changes along the way, especially with the dataTesters being more of a long-term project.

File: jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java (hunk 4 above)
Comment (pcwhite) [reply jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java:144]: @stiemannkj1 Thanks!

File: jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java (hunk 2 above)
Comment (pcwhite) [reply jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java:258]: @stiemannkj1 hmm, yea. missed that.

File: jsf-showcase-webapp/pom.xml (hunk 1 above)
Comment (ngriffin7a) [reply jsf-showcase-webapp/pom.xml:241]: Fixed in code review commit.

File: jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java (hunk 3 above)
Comment (ngriffin7a) [reply jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java:48]: Fixed in code review commit.

File: jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java (hunk 2 above)
Comment (ngriffin7a) [reply jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java:258]: Ran Jalopy in code review commit.

**Comment by ngriffin7a** [comment]
fp to 4.x and bp to 2.x <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by ngriffin7a** [comment]
Merged, thank you. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
